core/src/test/java/org/bitcoinj/utils/FiatTest.java

```diff
@@ -105,6 +105,7 @@ public void testValueFetching() {
         Fiat fiat = parseFiat("USD", "666");
         assertEquals(6660000, fiat.longValue());
         assertEquals("6660000", fiat.toString());
+        assertEquals(206545287, fiat.hashCode());
```

Comment:
This seems like overtesting to me. It's not important that `.hashCode()` yields a specific value – only that it stays at the same value if the object stays "equal".

At the very least, such a test should go to its own test method. It has got nothing to do with "value fetching".